client/src/App.tsx

```diff
@@ -33,7 +40,21 @@ function App() {
     fetchList();
   }, []);
 
-  const runAllTests = useCallback(
+  useEffect(() => {
+    const changeOptionalTestStatuses = () => {
+      setTestList(testList.map((test) => {
+        let op_status = runOptionalTests ? TestStatus.PENDING : TestStatus.SKIPPED;
+        if (test.name.includes("optional")) {
+          test.status = op_status;
+        };
+        return test;
+      }));
+    };
+    changeOptionalTestStatuses()
+  // eslint-disable-next-line
```

Comment:
whats this eslint warning?

Reply:
That I didn't have `testList` in the list of dependencies. But if I add it, the test list doesn't render. This effect doesn't need to fire when testList changes anyway since its initialized and then never changed.